react/features/local-recording/components/LocalRecordingInfoDialog.js

```diff
@@ -0,0 +1,332 @@
+
+    /**
+     * The recording duration string to be displayed on the UI.
+     */
+    durationString: string
+}
+
+/**
+ * A React Component with the contents for a dialog that shows information about
+ * local recording. For users with moderator rights, this is also the "control
+ * panel" for starting/stopping local recording on all clients.
+ *
+ * @extends Component
+ */
+class LocalRecordingInfoDialog extends Component<Props, State> {
+
+    /**
+     * Saves a handle to the timer for UI updates,
+     * so that it can be cancelled when the component unmounts.
+     */
+    _timer: ?IntervalID;
+
+    /**
+     * Constructor.
+     */
+    constructor() {
+        super();
+        this.state = {
+            durationString: 'N/A'
+        };
+    }
+
+    /**
+     * Implements React's {@link Component#componentWillMount()}.
+     *
+     * @returns {void}
+     */
+    componentWillMount() {
+        this._timer = setInterval(
+            () => {
+                this.setState((_prevState, props) => {
+                    const nowTime = new Date(Date.now());
+
+                    return {
+                        durationString: this._getDuration(nowTime,
+                            props.recordingStartedAt)
+                    };
+                });
+                try {
+                    this.props.dispatch(
+                        statsUpdate(recordingController
+                            .getParticipantsStats()));
+                } catch (e) {
+                    // do nothing
+                }
+            },
+            1000
+        );
+    }
+
+    /**
+     * Implements React's {@link Component#componentWillUnmount()}.
+     *
+     * @returns {void}
+     */
+    componentWillUnmount() {
+        if (this._timer) {
+            clearInterval(this._timer);
+            this._timer = null;
+        }
+    }
+
+
+    /**
+     * Returns React elements for displaying the local recording stats of
+     * each participant.
+     *
+     * @returns {ReactElement}
+     */
+    renderStats() {
+        const { stats, t } = this.props;
+
+        if (stats === undefined) {
+            return <ul />;
+        }
+        const ids = Object.keys(stats);
+
+        return (
+            <ul>
+                {ids.map((id, i) =>
+
+                    // FIXME: a workaround, as arrow functions without `return`
+                    // keyword need to be wrapped in parenthesis.
+                    /* eslint-disable no-extra-parens */
+                    (<li key = { i }>
+                        <span>{stats[id].displayName || id}: </span>
+                        <span>{stats[id].recordingStats
+                            ? `${stats[id].recordingStats.isRecording
+                                ? t('localRecording.clientState.on')
+                                : t('localRecording.clientState.off')} `
+                            + `(${stats[id]
+                                .recordingStats.currentSessionToken})`
```

Comment:
I see this is set to -1 by default. Will it make sense to see the -1? What does knowing the session token do for the user?

Reply:
Do remote users have currentSessionToken defined?

Reply:
- The reason for `-1` is that I just want something that is not a valid session id to indicate that there has been no recording sessions, since all session ids are non-negative numbers.

- Only the moderator generates "new" sessionTokens and sends them with START_COMMAND.  Remote users (non-moderators), upon receiving the START_COMMAND, will update their local `currentSessionToken` with the one received. 

- Essentially `currentSessionToken` functions like a revision id of `START_COMMAND` to make sure the states of each users is synced with the moderater's.